alfresco-transformer-base/src/test/java/org/alfresco/transformer/AbstractTransformerControllerTest.java

```diff
@@ -163,14 +169,35 @@ void generateTargetFileFromResourceFile(String actualTargetExtension, File testF
 
     protected File getTestFile(String testFilename, boolean required) throws IOException
     {
+        File testFile = null;
         ClassLoader classLoader = getClass().getClassLoader();
         URL testFileUrl = classLoader.getResource(testFilename);
         if (required && testFileUrl == null)
         {
             throw new IOException("The test file " + testFilename +
                     " does not exist in the resources directory");
         }
-        return testFileUrl == null ? null : new File(URLDecoder.decode(testFileUrl.getPath(), "UTF-8"));
+        if(testFileUrl!=null)
+        {
+            try 
+            {
+                testFile = folder.newFile(testFilename);
+                Files.copy(classLoader.getResourceAsStream(testFilename), testFile.toPath(),REPLACE_EXISTING);
+            } 
+            catch (IOException e) 
+            {
+                if(e.getMessage().contains("a file with the name \'" + testFilename + "\' already exists in the test folder"))
+                {
+                    testFile = new File(URLDecoder.decode(folder.getRoot().getPath()+ File.separator + testFilename, "UTF-8"));
+                }
+                else
+                {
+                    throw e;
+                }
+            }
+        }
+
```

Comment:
Allows files to be inherited test files to be read from the jars, so that duplicated resources are not required.

Reply:
Maybe add this reason as a comment to the code above ?

Reply:
Added comments